src/cryptmd5.ts

```diff
@@ -66,6 +66,17 @@ export class Hash {
     constructor(public value: string) { }
 }
 
+
+export function janus_hash(password: string, salt: string): Hash {
+    if('' === password) {
+        return new Hash('');
+    } else {
+        return crypt_md5(password, salt);
+    }
+}
+
+
+
 /**
```

Comment:
Ok, the good thing about that is, that the types are correct. No "new Hash('')" anymore. Plus it shows the user that there is something special about empty passwords. I think I would even add a comment to the documentation.
The disadvantage is, that I still have to make the decision about the type on my side before calling changeUser(). Actually I wanted to get rid of that decision... However, I think it's ok. It's probably better that way.